Sources/Bond/Collections/ObservableArray.swift

```diff
@@ -295,30 +296,30 @@ extension MutableObservableArray {
 }
 
 extension MutableObservableArray where Item: Equatable {
-  
+
   public func replace(with array: [Item], performDiff: Bool) {
-    if performDiff {
-      lock.lock()
+    lock.lock(); defer { lock.unlock() }
 
+    if performDiff {
       let diff = self.array.extendedDiff(array)
+      let patch = diff.patch(from: self.array, to: array)
```

Comment:
I think you could simplify the `func patchedBatch()` method to something like `var patches`, or `func toPatches()`.

In terms of the `ObservableArrayEvent` naming - perhaps `ObservableArrayDiffEvent` (replacing `ObservableArrayEvent`) and `ObservableArrayPatchEvent` for the new variant?